lib/teleagent/lazy.go

```diff
@@ -0,0 +1,112 @@
+package teleagent
+
+import (
+	"context"
+	"sync"
+
+	"github.com/gravitational/trace"
+
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/agent"
+)
+
+// lazyAgent is an agent.Agent instance which is lazily initialized
+// via a call to an AgentGetter.  Helpful for dealing with APIs
+// which require, but may not actually use, an agent.
+type lazyAgent struct {
+	ctx  context.Context
```

Comment:
same as above: https://github.com/golang/go/wiki/CodeReviewComments#contexts
pass context through func arguments instead of struct fields